spec.bs

```diff
@@ -1387,6 +1478,227 @@ To <dfn>build an interest group passed to generateBid</dfn> given an [=interest
   1. Return |igGenerateBid|.
 </div>
 
+<div algorithm>
+To <dfn>validate and convert additional bids</dfn> given an [=auction config=] |auctionConfig|, an
+[=auction config=]-or-null |topLevelAuctionConfig|, and a [=map=] |negativeTargetInfo| whose
+[=map/keys=] are [=tuples=] consisting of an [=origin=] and a [=string=], and whose [=map/values=]
+are [=tuples=] consisting of an [=origin=] and a [=byte sequence=]:
+
+  1. Let |auctionNonce| be |auctionConfig|'s [=auction config/auction nonce=].
+  1. Let |capturedAdditionalBidsHeaders| be [=this=]'s [=relevant global object=]'s
+    [=associated Document's=] [=node navigable's=] [=traversable navigable's=]
+    [=captured additional bids headers=].
+  1. Let |additionalBids| be a new [=list=] of [=additional bids=].
+  1. [=list/For each=] |encodedSignedAdditionalBid| of |capturedAdditionalBidsHeaders|[|auctionNonce|]:
+    1. Let |signedAdditionalBid| be the result of running [=forgiving-base64 decode=] with
+      |encodedSignedAdditionalBid|.
+    1. If |signedAdditionalBid| is failure, then [=iteration/continue=].
+    1. Let |additionalBid| be the result of running [=parse a signed additional bid=] given
+      |signedAdditionalBid|, |auctionConfig|, |topLevelAuctionConfig|, and |negativeTargetInfo|. 
+    1. If |additionalBid| is not null:
+      1. [=list/Append=] |additionalBid| to |additionalBids|.
+  1. Return |additionalBids|.
+</div>
+
+<div algorithm>
+To <dfn>parse a signed additional bid</dfn> given a [=byte sequence=] |signedAdditionalBid|, an
+[=auction config=] |auctionConfig|, an [=auction config=]-or-null |topLevelAuctionConfig|, and a
+[=map=] |negativeTargetInfo| whose [=map/keys=] are [=tuples=] consisting of an [=origin=] and a
+[=string=], and whose [=map/values=] are [=tuples=] consisting of an [=origin=] and a
+[=byte sequence=]:
+
+  1. Let |parsedSignedAdditionalBid| be the result of [=parse a JSON string to an infra value=]
+    given |signedAdditionalBid|.
+  1. Return null if any of the following conditions hold:
+    * |parsedSignedAdditionalBid| is not a [=map=];
+    * |parsedSignedAdditionalBid|["bid"] does not [=map/exist=], or it's not a [=string=];
+    * |parsedSignedAdditionalBid|["signatures"] does not [=map/exist=], or it's not a [=list=].
+  1. Let |signatures| be a new [=list=] of [=signed additional bid signatures=].
+  1. Let |decodeSignatureFailed| be false.
+  1. [=list/For each=] |sig| of |parsedSignedAdditionalBid|["signatures"]:
+    1. Set |decodeSignatureFailed| to true and [=iteration/break=] if any of the following
+      conditions hold:
+      * |sig| is not a [=map=];
+      * |sig|["key"] does not [=map/exist=], or it's not a [=string=];
+      * |sig|["signature"] does not [=map/exist=], or it's not a [=string=].

```

Comment:
"is" instead of "it's". I'm spotting this below in a few more algorithms, so I'll just use this comment as a place to track the remaining instances.